berkeley-function-call-leaderboard/bfcl_eval/model_handler/api_inference/ling.py

```diff
@@ -0,0 +1,179 @@
+
+class LingAPIHandler(BaseHandler):
+    def __init__(self, model_name, temperature) -> None:
+        super().__init__(model_name, temperature)
+        self.model_style = ModelStyle.Ling
+        self.model_name = model_name
+        self.is_fc_model = False
+        self.temperature = temperature
+        api_url="https://bailingchat.alipay.com"
+        self.client = OpenAI(
+            base_url=api_url,
+            api_key=os.getenv("Ling_API_KEY")
+        )
+    
+    def decode_ast(self, result, language="Python"):
+        if "FC" not in self.model_name:
+            pattern = r"\[.*\]"
+            m = re.match(pattern, result)
+            if m:
+                func=m.group()
+            else:
+                func=result
+            decode_output = ast_parse(func, language)
+            return decode_output
+        else:
+            decoded_output = []
+            for invoked_function in result:
+                name = list(invoked_function.keys())[0]
+                params = json.loads(invoked_function[name])
+                decoded_output.append({name: params})
+            return decoded_output
+        
+    def decode_execute(self, result):
+        if "FC" not in self.model_name:
+            pattern = r"\[.*\]"
+            m = re.match(pattern, result)
+            if m:
+                func=m.group()
+            else:
+                func=result
+            decode_output = ast_parse(func)
+            execution_list = []
+            for function_call in decode_output:
+                for key, value in function_call.items():
+                    execution_list.append(
+                        f"{key}({','.join([f'{k}={repr(v)}' for k, v in value.items()])})"
+                    )
+            return execution_list
+        else:
+            function_call = convert_to_function_call(result)
+            return function_call
+         
+    #### Prompting methods ####
+    def _query_prompting(self, inference_data: dict):
+        """
+        Call the model API in prompting mode to get the response.
+        Return the response object that can be used to feed into the decode method.
+        """
+        prompt = ""
+        if "system_message" in inference_data:
+            prompt+="<role>SYSTEM</role>"+inference_data["system_message"]
+        for message in inference_data["messages"]:
+            if message["role"]=="user":
+                prompt+="<role>HUMAN</role>"+message["content"]
+            elif message["role"] == "assistant":
+                prompt+="<role>ASSISTANT</role>"+message["content"]
+            elif message["role"]=="function":
+                prompt+="<role>OBSERVATION</role>"+message["content"]
+        prompt+="<role>ASSISTANT</role>"
+        messages=[
+            {
+                "role":"system",
+                "content":"You are a helpful assistant."
+            },
+            {
+                "role":"user",
+                "content":prompt
```

Comment:
Could you clarify why the handler wraps all messages into a single user prompt, instead of directly passing `inference_data["messages"]` to the chat completion endpoint (i.e., `messages=inference_data["messages"]`)?

Reply:
Thanks for making the changes!